### PR TITLE
Add "inverted" to PluginRequirement

### PR DESCRIPTION
**Pull request by mdellweg**
Also remove the deprecated interface from has_plugin and needs_plugin.

[noissue]

This should help to implement #261 .

### CODE REVIEW

File: pulpcore/cli/common/context.py [hunk 1]
```diff
@@ -281,13 +223,19 @@ def needs_plugin(
                 if plugin.max is not None:
                     specifier += f"{separator}<{plugin.max}"
                 feature = plugin.feature or _("this command")
-                raise click.ClickException(
-                    _(
+                if plugin.inverted:
+                    msg = _(
+                        "The server provides the pulp component '{specifier}',"
+                        " which prevents to use {feature}."
```
Comment (ggainey) [inline pulpcore/cli/common/context.py:229]: ^to use^the use of

**Review by ggainey — CHANGES_REQUESTED** [review]
One minor working-tweak, otherwiuse lg2m

File: docs/architecture.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ To facilitate this, the `PulpContext` provides the `needs_plugin` and `has_plugi
 ```python
 class MyEntityContext(PulpEntityContext):
     def show(self, href):
-        if not self.pulp_ctx.has_plugin("pulp_my_content", min_version="1.2.3"):
+        if not self.pulp_ctx.has_plugin(PluginRequirement("pulp_my_content", min="1.2.3")):
```
Comment (daviddavis) [inline docs/architecture.md:60]: AFAICT, these changes are not related to the inverted feature that you're also introducing here. Is that correct?

**Comment by daviddavis** [comment]
Might be worth adding some information about this inverted feature here:

https://github.com/pulp/pulp-cli/blob/develop/docs/architecture.md#version-dependent-code-paths

File: docs/architecture.md (hunk 2 above)
Comment (mdellweg) [reply docs/architecture.md:60]: In a way. Those functions have the "old" interface aka "passing parameters individually" and the new one, "passing the `PluginRequrement` as a single object".
The old interface was meant to go away anyway, and i didn't bother to add the `inverted` to it. So i decided for removing it here.
Why do we have the sligtly more complex "new" interface? `needs_plugin` works deferred, and it was very convenient to store and pass around the `PluginRequirement` objects.

**Review by ggainey — APPROVED** [review]

**Review by daviddavis — APPROVED** [review]
